pkg/providers/cloudstack/config/template-cp.yaml

```
@@ -58,50 +74,59 @@ kind: CloudStackMachineTemplate
 metadata:
   name: {{.controlPlaneTemplateName}}
   namespace: {{.eksaSystemNamespace}}
-{{- if .cloudstackControlPlaneAnnotations}}
+{{- if $annotationsCP}}
   annotations:
 {{- end }}
-{{- if .cloudstackControlPlaneDiskOfferingProvided}}
-    mountpath.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingPath}}
-    device.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingDevice}}
-    filesystem.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingFilesystem}}
-    label.diskoffering.{{.cloudstackAnnotationSuffix}}: {{.cloudstackControlPlaneDiskOfferingLabel}}
+{{- if $diskConfiguredCP}}
+    diskoffering.{{$suffix}}: "{{.cloudstackControlPlaneDiskOfferingJsonStr | js}}"
+{{- end }}
+{{- if $isoConfiguredCP}}
+    ISOAttachment.{{$suffix}}: "{{.cloudstackControlPlaneISOAttachmentJsonStr | js}}"
 {{- end }}
 {{- if .cloudstackControlPlaneSymlinks }}
-    symlinks.{{.cloudstackAnnotationSuffix}}: {{ $first := true }}{{- range $key, $value := .cloudstackControlPlaneSymlinks}}{{if not $first}},{{else}}{{$first = false}}{{- end}}{{$key}}:{{$value}}{{- end}}
+    symlinks.{{$suffix}}: "{{.cloudstackControlPlaneSymlinksJsonStr | js}}"
 {{- end}}
 spec:
   template:
     spec:
       offering:
-{{- if .cloudstackControlPlaneComputeOfferingId}}
-        id: {{.cloudstackControlPlaneComputeOfferingId}}
+{{- if $offeringCP.Id}}
+        id: {{$offeringCP.Id}}
 {{- end}}
```

do we need these `if`'s? Or can we just always include this line and have an empty value? Same for the other conditional lines

```suggestion
        id: {{$offeringCP.Id}}
```

---

technically, we don't need these if -- the generated yaml may looks tidier if we have if.
I agree if we can remove lots of if for id/name in diskoffering, compute offering, template, isoattachment.